packages/react-widgets/src/models/CategoryModel.js

```diff
@@ -1,22 +1,55 @@
+import { executeSQL } from '@carto/react-api/';
+import { AggregationTypes } from '@carto/react-core/';
 import { Methods, executeTask } from '@carto/react-workers';
+import {
+  formatOperationColumn,
+  formatTableNameWithFilters,
+  wrapModelCall
+} from './utils';
 
-export const getCategories = async (props) => {
-  const {
-    column,
-    operationColumn,
-    operation,
-    joinOperation,
-    filters,
-    filtersLogicalOperator,
-    dataSource
-  } = props;
-
-  return executeTask(dataSource, Methods.FEATURES_CATEGORY, {
-    filters,
-    filtersLogicalOperator,
+export function getCategories(props) {
+  return wrapModelCall(props, fromLocal, fromRemote);
+}
+
+// From local
+function fromLocal(props) {
+  const { source, column, operationColumn, operation, joinOperation } = props;
+
+  return executeTask(source.id, Methods.FEATURES_CATEGORY, {
+    filters: source.filters,
+    filtersLogicalOperator: source.filtersLogicalOperator,
     operation,
     joinOperation,
     column,
     operationColumn: operationColumn || column
   });
-};
+}
+
+// From remote
+function fromRemote(props) {
+  const { source, abortController } = props;
+  const { credentials, connection } = source;
+
+  const query = buildSqlQueryToGetCategories(props);
+
+  return executeSQL({
+    credentials,
+    query,
+    connection,
+    opts: { abortController }
+  });
+}
+
+function buildSqlQueryToGetCategories(props) {
+  const { column, operation, operationColumn, joinOperation } = props;
+
+  const selectValueClause = `${operation}(${
+    operation === AggregationTypes.COUNT
+      ? '*'
+      : formatOperationColumn(operationColumn || column, joinOperation)
+  }) as value`;
+
+  return `SELECT COALESCE(${column}, 'null') as name, ${selectValueClause} FROM ${formatTableNameWithFilters(
```

Comment:
Just to confirm, what warehouses should support this feature?

Reply:
Any of them. Do you know any incompatibility here?

Reply:
Nop, but it would be nice to involve Data team in these queries verification. Can you gather them (not just this one) and pass them to @Jesus89 please?